component/discovery/file/file.go

```diff
@@ -116,239 +77,56 @@ func (c *Component) Update(args component.Arguments) error {
 	defer c.mut.Unlock()
 
 	c.args = args.(Arguments)
-	c.reconcileWatchesWithWatcher()
+	c.watches = c.watches[:0]
+	for _, v := range c.args.PathTargets {
+		c.watches = append(c.watches, watch{
+			target: v,
+		})
+	}
 	return nil
 }
 
 // Run satisfies the component interface.
 func (c *Component) Run(ctx context.Context) error {
-	watchDog := time.NewTicker(c.args.UpdatePeriod)
-	timerDuration := c.args.UpdatePeriod
+	watchDog := time.NewTicker(c.args.SyncPeriod)
+	timerDuration := c.args.SyncPeriod
 	update := func() {
 		c.mut.Lock()
 		defer c.mut.Unlock()
 
-		// See if there is anything new we need to check.
-		c.reconcileWatchesWithWatcher()
-		// Update the exports with the targets. Should only be called if changes occurred.
-		c.checkOnStateChanged()
+		paths := c.getWatchedFiles()
+		// The component node checks to see if exports have actually changed.
+		c.opts.OnStateChange(Exports{Targets: paths})
+
 		// Check to see if our ticker timer needs to be reset.
-		if timerDuration != c.args.UpdatePeriod {
-			watchDog.Reset(c.args.UpdatePeriod)
-			timerDuration = c.args.UpdatePeriod
+		if timerDuration != c.args.SyncPeriod {
+			watchDog.Reset(c.args.SyncPeriod)
+			timerDuration = c.args.SyncPeriod
```

Comment:
Do we need to reset the ticker timer in the Update method instead?

I think that here, if we had an initial SyncPeriod of 1h, and reloaded config to change it to 20s, the change would only take place after the 1h has elapsed first, right?

Reply:
That makes sense